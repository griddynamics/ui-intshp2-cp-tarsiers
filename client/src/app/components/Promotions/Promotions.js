import React, { Component } from 'react';
import appConfig from '../../../config/appConfig';
import Slider from '../Slideshow/Slider';
import Spinner from '../../shared/Spinner';

export default class Promotions extends Component {
  constructor(props) {
    super(props);
    this.state = { slides: [] };
  }

  componentDidMount = () => {
    setTimeout(
      () => this.setState({ slides: [...appConfig.promotions.slides] }),
      1500
    );
  };

  get isLoaded() {
    const { slides } = this.state;

    return slides.length;
  }

  render() {
    const { slides } = this.state;
    const style = { height: '40vw' };

    return (
      <section className="promotions" style={style}>
        {this.isLoaded ? <Slider slides={slides} /> : <Spinner />}
      </section>
    );
  }
}
