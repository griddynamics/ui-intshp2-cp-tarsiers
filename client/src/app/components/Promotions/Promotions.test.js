import React from 'react';
import { shallow } from 'enzyme';
import Promotions from './Promotions';

describe('<Promotions />', () => {
  it('should match its snapshot', () => {
    const props = { slides: [] };
    const wrapper = shallow(<Promotions {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
