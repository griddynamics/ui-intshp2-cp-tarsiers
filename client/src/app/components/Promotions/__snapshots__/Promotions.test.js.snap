// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Promotions /> should match its snapshot 1`] = `
<section
  className="promotions"
  style={
    Object {
      "height": "40vw",
    }
  }
>
  <Spinner />
</section>
`;
